pos_product_cost_security/models/product_product.py

```diff
@@ -0,0 +1,18 @@
+# Copyright 2021 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from odoo import models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def read(self, fields=None, load="_classic_read"):
+        """Make an exception when loading data to PoS"""
+        if self.env.context.get(
+            "pos_override_cost_security"
+        ) and self.env.user.has_group("point_of_sale.group_pos_user"):
+            fields = fields if fields else []
+            other_fields = [f for f in fields if f != "standard_price"]
+            self.check_field_access_rights("read", other_fields)
+            return super(ProductProduct, self.sudo()).read(fields=fields, load=load)
```

Comment:
Hummm 🤔 OK but... if the request is asking for that field, couldn't it be because it's needed? Or is it just that the PoS just downloads all fields always?

Also, wouldn't it be a cleaner fix would be to add `groups` to the `standard_price` field?

Reply:
Well, the PoS loads very specific fields. The only one in conflict is `standard_price`, but is a field needed to compute prices if a cost based pricelist is there.

We can't give access to the PoS users, as they don't necessarily have to own such permission.

So the thing is that we need to load the data for the PoS to work (remember that all the logic is reimplemented in the frontend)